src/components/wrappers/in-quiz-wrapper.tsx

```diff
@@ -1,13 +1,48 @@
-import { ReactNode } from 'react';
+import { ReactNode, useEffect } from 'react';
+import { useRouter } from 'next/router';
 import QuizHead from '../headers/quiz-head';
 import { Wrapper } from './wrapper';
 import type { Props as WrapperProps } from './wrapper';
+import { useTeamQuizDataStore } from '../../state/team/team-quiz-data.state';
+import { trpc } from '../../utils/trcp';
+import { QuizStatusEnum } from '.prisma/client';
 
 interface Props extends WrapperProps {
   children: ReactNode;
+  quizStatus?: QuizStatusEnum;
 }
 
-export default function InQuizWrapper({ children, ...props }: Props) {
+export default function InQuizWrapper({
+  children,
+  quizStatus,
+  ...props
+}: Props) {
+  const router = useRouter();
+
+  const { setQuizData, quizData } = useTeamQuizDataStore((state) => ({
+    setQuizData: state.setQuizData,
+    quizData: state.quizData,
+  }));
+
+  useEffect(() => {
+    if (!quizData.pin) {
+      router.push('/pin');
+    }
+  }, []);
+
+  trpc.team.getQuizByPin.useQuery(
+    {
+      pin: quizData.pin!,
+      quizStatus: quizStatus || QuizStatusEnum.PLAYING,
```

Comment:
Why is it not enough to just query by pin?

Reply:
BE called is reused in couple of places, Initially I wanted to have it only for pin screen, where you get quiz which is in state of Joining, but during this PR decided to make it a bit more general, to have status more flexible.
And to answer your question, you can imagine that there will be more quizes with the same pin but different statuses. I know It does not solve uniqueness problem completely, but for now it's good enough